plugins/modules/efs_tag.py

```diff
@@ -0,0 +1,174 @@
+    default: present
+    choices: ['present', 'absent']
+    type: str
+  tags:
+    description:
+      - A dictionary of tags to add or remove from the resource.
+      - If the value provided for a tag is null and I(state=absent), the tag will be removed regardless of its current value.
+    type: dict
+  purge_tags:
+    description:
+      - Whether unspecified tags should be removed from the resource.
+      - Note that when combined with I(state=absent), specified tags with non-matching values are not purged.
+    type: bool
+    default: false
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = r'''
+- name: Ensure tags are present on a resource
+  community.aws.efs_tag:
+    resource: fs-123456ab
+    state: present
+    tags:
+      Name: MyEFS
+      Env: Production
+
+- name: Remove the Env tag if it's currently 'development'
+  community.aws.efs_tag:
+    resource: fsap-78945ff
+    state: absent
+    tags:
+      Env: development
+
+- name: Remove all tags except for Name
+  community.aws.efs_tag:
+    resource: fsap-78945ff
+    state: absent
+    tags:
+        Name: foo
+    purge_tags: true
+'''
+
+RETURN = r'''
+tags:
+  description: A dict containing the tags on the resource
+  returned: always
+  type: dict
+added_tags:
+  description: A dict of tags that were added to the resource
+  returned: If tags were added
+  type: dict
+removed_tags:
+  description: A dict of tags that were removed from the resource
+  returned: If tags were removed
+  type: dict
+'''
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    # Handled by AnsibleAWSModule
+    pass
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict, ansible_dict_to_boto3_tag_list, compare_aws_tags
+
+
+def get_tags(efs, module, resource):
+    '''
+    Get resource tags
+    '''
+    try:
+        return boto3_tag_list_to_ansible_dict(efs.list_tags_for_resource(ResourceId=resource)['Tags'])
+    except (BotoCoreError, ClientError) as get_tags_error:
+        module.fail_json_aws(get_tags_error, msg='Failed to fetch tags for resource {0}'.format(resource))
+
+
+def main():
+    '''
+    MAIN
+    '''
+    argument_spec = dict(
+        resource=dict(required=False),
+        tags=dict(type='dict'),
+        purge_tags=dict(type='bool', default=False),
+        state=dict(default='present', choices=['present', 'absent'])
+    )
+
+    required_if = [('state', 'present', ['tags']), ('state', 'absent', ['tags'])]
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, required_if=required_if, supports_check_mode=True)
+    resource = module.params['resource']
+    tags = module.params['tags']
+    state = module.params['state']
+    purge_tags = module.params['purge_tags']
+
+    result = {'changed': False}
+
+    efs = module.client('efs')
```

Comment:
```suggestion
    efs = module.client('efs', retry_decorator=AWSRetry.jittered_backoff())
```

It's generally worth adding a retry decorator here, and then adding aws_retry=True to the add/remove list calls